third_party/move/move-compiler-v2/src/env_pipeline/lambda_lifter.rs

```diff
@@ -232,10 +233,8 @@ impl<'a> LambdaLifter<'a> {
         let mut params = vec![];
         let mut saw_error = false;
 
-        for (used_param_count, (param, var_info)) in
-            mem::take(&mut self.free_params).into_iter().enumerate()
-        {
-            let name = self.fun_env.get_local_name(param);
+        for (used_param_count, (param, var_info)) in self.free_params.iter().enumerate() {
```

Comment:
I think this also fixes https://github.com/aptos-labs/aptos-core/issues/16199. Can you please add that test case too and make this PR close it?